drivers/display/display_st7789v.c

```diff
@@ -101,16 +99,11 @@ static void st7789v_exit_sleep(struct st7789v_data *data)
 static void st7789v_reset_display(struct st7789v_data *data)
 {
 	LOG_DBG("Resetting display");
-#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
 	k_sleep(K_MSEC(1));
 	gpio_pin_set(data->reset_gpio, ST7789V_RESET_PIN, 1);
 	k_sleep(K_MSEC(6));
 	gpio_pin_set(data->reset_gpio, ST7789V_RESET_PIN, 0);
 	k_sleep(K_MSEC(20));
-#else
-	st7789v_transmit(p_st7789v, ST7789V_CMD_SW_RESET, NULL, 0);
```

Comment:
This software reset does not work or what is reasoning?

Reply:
@jfischer-no Well, on `main` branch it is currently not possible to enable software reset. To enable software reset you need to leave out the reset pin, however defining a reset pin is required. This indicated to me that nobody needed software reset and that is was never tested. I thought this change would simplify the code.

Reply:
@caspermeijn can you remove the `required` field from the reset-gpios? Then you can obtain the gpio using `GPIO_DT_SPEC_INST_GET_OR(0, reset_gpios, {0})`. This way you can then add both, hard and soft reset options.

Reply:
@gmarull I will maintain the status-quo in this PR and open a separate PR for the software reset.

Reply:
I think it's better to address it here (in a separate commit), the only issue with the current status is the `required: True` field in the dt binding.

Reply:
Added a commit to fix software reset